students/nDruP/lesson04/mailroom2.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+"""
+1. Create Data structure that holds Donor, Donation Amount.
+2. Prompt user to Send a Thank You, Create a Report, or quit.
+3. At any point, the user should be able to quit their current task and return
+to the original prompt
+4. From the original prompt, the user should be able to quit the scipt cleanly
+"""
+
+
+import sys
+import os
+import datetime
+
+
+donor_dict = {"Sleve McDichael": [86457.89, 2346.43, 9099.09],
+              "Willie Dustice": [505.05, 43.21],
+              "Rey McScriff": [666.00],
+              "Mike Truk": [70935.30, 12546.70, 312.00],
+              "Bobson Dugnutt": [1234.56, 789.00],
+              "Todd Bonzalez": [715867.83, 10352.07]}
+
+
+def main_menu():
+    """
+    Prompt user to send a Thank You, Create a report, create letters, or quit.
+    """
+    user_prompt = None
+    valid_prompts = {"1": craft_thank_u,
+                     "2": create_report,
+                     "3": create_letters,
+                     "4": sys.exit}
+    while True:
+        print_divider()
+        print("We're a Pyramid Scheme & So Are You! E-Mailroom")
+        print_divider()
+        while user_prompt not in valid_prompts:
+            print("Please choose from the following options (1,2,3,4): ")
```

Comment:
could use string format here for options and use values from `valid_promts.keys()`